swath_projector/reproject.py

```diff
@@ -122,14 +124,13 @@ def get_parameters_from_message(
     if parameters['interpolation'] in [None, '', 'None']:
         parameters['interpolation'] = INTERPOLATION_DEFAULT
 
-    # ERROR 5: -tr and -ts options cannot be used at the same time.
+    # when a user requests both a resolution and dimensions, then ensure the
+    # extents are consistent.
     if (parameters['xres'] is not None or parameters['yres'] is not None) and (
         parameters['height'] is not None or parameters['width'] is not None
     ):
```

Comment:
What is the behaviour when only one of the dimensions (x or y) has both the resolution and the height or width defined?

I think there's maybe something to think about [here](https://github.com/nasa/harmony-swath-projector/blob/main/swath_projector/interpolation.py#L498), [here](https://github.com/nasa/harmony-swath-projector/blob/main/swath_projector/interpolation.py#L397) and [here](https://github.com/nasa/harmony-swath-projector/blob/main/swath_projector/interpolation.py#L435), because I think now it is possible to have a self consistent grid that has (for example) `scaleExtent`, `scaleSize` and then only one of `height` and/or `width`.

I _think_, with these changes, it technically makes no difference - what would happen is that the code would calculate the missing value (say `width`) and then also calculate `height`. And because the grid is self-consistent, the value of `height` would come out the same, but it feels a bit clunky to recalculate something we already have.

I guess the conclusion is: the code isn't doing anything wrong, or getting a bad value, but there's a coupling between the x and y grid parameters that means the code doesn't quite make as much sense in the case outlined. Feel free to tell me this doesn't matter too much, because it ends up all the same, but it feels like maybe it's better to decouple x and y to make things easier to grok.

Reply:
I'm not 100% sure I understand you fully.

From what I see this function is called at the beginning of [reproject()](https://github.com/nasa/harmony-swath-projector/blob/mhs%2FDAS-1934%2Fvalidate-grid-messages-correctly/swath_projector/reproject.py#L26) inside [adaper's process_item](https://github.com/nasa/harmony-swath-projector/blob/mhs%2FDAS-1934%2Fvalidate-grid-messages-correctly/swath_projector/adapter.py#L79)

> What is the behaviour when only one of the dimensions (x or y) has both the resolution and the height or width defined?

Then if the user only gave a single value `height` or `width`, it would wipe out whichever one was input [here](https://github.com/nasa/harmony-swath-projector/blob/mhs%2FDAS-1934%2Fvalidate-grid-messages-correctly/swath_projector/interpolation.py#L410) and recalculate both.
Same thing would happen if a single resolutions was input with valid dimensions and extents.

> It feels a bit clunky to recalculate something we already have.

True, but I'm a big meh about it.


> there's a coupling between the x and y grid parameters that means the code doesn't quite make as much sense in the case outlined.

I feel like you're trying to get at something with this, but I don't know what it is.  The only difference I can see is that we don't automatically fail when you over describe your input paramters.  Adding an single extra dimension that matches the input is kinda dump but it doesn't break anything, and if you provide both we don't do any recomputations (beyond the ones we did to verify the grid)

So I need more information but leaning towards a big shrug.

Reply:
Looking more closely [here](https://github.com/nasa/harmony-swath-projector/blob/main/swath_projector/interpolation.py#L460), I guess that decoupling x and y is perhaps more tricky than I'd first thought. Perhaps, I'm backing down a little bit here on my initial thoughts: yes it feels clunky/unintuitive to couple x and y the way they are now, but maybe there isn't a clean way to separate them in some cases.

Maybe I'm beginning to also say :shrug: